### DISCUSSION TITLE
Allow the disabling of Snapshot expireOptions "Never" selection

### DISCUSSION THREAD

**Discussion by IntegersOfK** [grafana/grafana#38201] [Dashboards, Panels & Visualizations]
My organization has thousands of users of a single Grafana instance, and I'd like to be able to invite people to use the Snapshots feature to essentially increase data retention for select queries that will not be available eventually from our APIs.

However, the current implementation of this dropdown allows the selection of "Never" expire Snapshots. I would like to be able to disable "Never," as we do not have enough database space to store the data indefinitely and my org is highly regulated in that users would assume their selection is going to be honoured.

This feature request is simply to allow the disabling of that "Never" option in the Snapshots expiry dropdown. Personally, I don't even think it's even necessary to enforce that from the backend, just allowing a setting to remove it from the frontend dropdown is enough for me.

Bonus: At the same time, I think it would be pretty harmless to add a few more expiry options. Currently [the options](https://github.com/grafana/grafana/blob/73518bf1e74d1017e2fd8bc8e478d1253ed751aa/public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx#L22-L27) are: 1 hour, 1 day, and 1 week. I propose adding 1 month, 3 months, and 1 year.

**Comment by marefr** [comment]
For a workaround you might be able to setup a cronjob or similar to cleanup very old snapshots.

**Comment by IntegersOfK** [comment]
Oh yes it's easy to delete old snapshots programmatically, but my users assume their selection of "Never" is going to be honoured.

Ie. I don't have the right to delete old snapshots if they chose "Never."

It's about user expectations. I won't be able to keep them forever, so I'd like to ensure they can't even select "Never."

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by minhaz1** [comment]
Any updates on this? I want to take it a step further and allow full customization of what options show up there. For example some people may want a 90 day or 1 yr expiry option.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
